funcr/funcr.go

```diff
@@ -596,6 +596,56 @@ func isEmpty(v reflect.Value) bool {
 	return false
 }
 
+func invokeMarshaler(m logr.Marshaler) (ret interface{}) {
+	defer func() {
+		if err := recover(); err != nil {
+			if isNil(m) {
+				ret = "<nil-logr-marshaler>"
+			} else {
+				panic(err)
```

Comment:
I think this should return a string explaining the problem, similar to what fmt.Sprint does.

We could also log an error, to raise awareness.